buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.kt

```diff
@@ -232,20 +249,72 @@ abstract class DackkaPlugin : Plugin<Project> {
         dependsOnAndMustRunAfter("createFullJarRelease")
 
         val dackkaFile = project.provider { project.dackkaConfig.singleFile }
-        val dackkaOutputDirectory = File(project.buildDir, "dackkaDocumentation")
 
         dackkaJarFile.set(dackkaFile)
-        outputDirectory.set(dackkaOutputDirectory)
         clientName.set(project.firebaseConfigValue { artifactId })
     }
 
-    private fun registerFiresiteTransformTask(project: Project, dackkaFilesDirectory: Provider<File>) =
-        project.tasks.register<FiresiteTransformTask>("firesiteTransform") {
-            mustRunAfter("generateDackkaDocumentation")
+    // TODO(b/248302613): Remove when dackka exposes configuration for this
+    private fun registerSeparateJavadocAndKotlinDoc(
+        project: Project,
+        dackkaOutputDirectory: Provider<File>,
+        outputDirectory: Provider<File>
+    ): TaskProvider<Task> {
+        val outputJavadocFolder = project.childFile(outputDirectory, "android")
+        val outputKotlindocFolder = project.childFile(outputDirectory, "kotlin")
+
+        val separateJavadoc = project.tasks.register<Copy>("separateJavadoc") {
+            dependsOn("generateDackkaDocumentation")
+
+            val javadocClientFolder = project.childFile(dackkaOutputDirectory, "reference/client")
+            val javadocComFolder = project.childFile(dackkaOutputDirectory, "reference/com")
+
+            fromDirectory(javadocClientFolder)
+            fromDirectory(javadocComFolder)
+
+            into(outputJavadocFolder)
+        }
+
+        val separateKotlindoc = project.tasks.register<Copy>("separateKotlindoc") {
+            dependsOn("generateDackkaDocumentation")
+
+            val kotlindocFolder = project.childFile(dackkaOutputDirectory, "reference/kotlin")
+
+            from(kotlindocFolder)
+
+            into(outputKotlindocFolder)
+        }
+
+        return project.tasks.register("separateJavadocAndKotlinDoc") {
+            dependsOn(separateJavadoc, separateKotlindoc)
+        }
+    }
+
+    private fun registerFiresiteTransformTask(
+        project: Project,
+        separatedFilesDirectory: Provider<File>,
+        targetDirectory: Provider<File>
+    ): TaskProvider<Task> {
+        val transformJavadoc = project.tasks.register<FiresiteTransformTask>("firesiteTransformJavadoc") {
+            dependsOnAndMustRunAfter("separateJavadoc")
+
+            referenceHeadTagsPath.set("google3/third_party/devsite/firebase/en/docs/reference/android")
```

Comment:
```suggestion
            referenceHeadTagsPath.set("en/docs/reference/android")
```
hm, I don't think we need the absolute path, but rather relative to the tenant(i.e. firebase)